### ISSUE TITLE
Add coverage of wait() support macros

### ISSUE THREAD

**Issue by mdoucha** [linux-test-project/ltp#605]
Since the statx05 test broke on s390 recently due to glibc bug in `wait()` support macros, we should check that those macros work correctly, otherwise dozens of tests which rely on subprocesses may miss regressions.

Macros to test:
- ``
- ``
- ``
- ``
- ``
- ``

**Comment by metan-ucw** [comment]
Some may be already covered by other tests.

For instance abort01 test exercise WCOREDUMP() as well as WTERMSIG() to some extend. The WIFSTOPPED() and WSTOPSIG() is tested in the waitpidXY tests.

But there is no trace of WIFCONTINUED() in the whole source tree, so I guess that a test that stops and resumes child and checks that wait returns corrects status is missing completely.

**Comment by mdoucha** [comment]
> Some may be already covered by other tests.
> 
> For instance abort01 test exercise WCOREDUMP() as well as WTERMSIG() to some extend. The WIFSTOPPED() and WSTOPSIG() is tested in the waitpidXY tests.

No, these are all tests which are at risk of being broken by buggy glibc macros. We need a test where a subprocess will (be) terminate(d) with predefined status and then check that the support macros parse the exit status correctly. Just using the macros to parse a possibly non-determinitic exit status is not a viable test.

**Comment by mcgrof** [comment]
> Since the statx05 test broke on s390 recently due to glibc bug in `wait()` support macros, 

Which bug was that exactly? I just git pulled glibc and I can't see any s390 specific type of bug for W*() macros.

**Comment by metan-ucw** [comment]
s390 had non-generic implemenation of these which didn't work with long. Arguably it wasn't a bug since you are supposed to pass only int pointer, but it was working fine on the rest of the architectures.

See also:

https://sourceware.org/bugzilla/show_bug.cgi?id=19613
